src/main/kotlin/assertk/assertions/inputstream.kt

```diff
@@ -0,0 +1,118 @@
+ */
+fun Assert<InputStream>.hasSameContentAs(other: InputStream) {
+
+    var size = 0L
+
+    val actualBuffer = ByteArray(BUFFER_SIZE)
+    val otherBuffer = ByteArray(BUFFER_SIZE)
+
+    while (true) {
+        val actualRead = fillBuffer(actual, actualBuffer)
+        val otherRead = fillBuffer(other, otherBuffer)
+
+        if (actualRead == otherRead) {
+
+            if(actualRead==0) {
+                // the streams have the same size and are finished and have the same content
+                return
+            }
+            else {
+                val pos = compare(actualRead, actualBuffer, otherBuffer)
+                if(pos==null) {
+                    // the current buffers are equals and we can try the next block
+                    size += actualRead
+
+                } else {
+                    // the first difference is in position pos
+
+                    val actualSize = size + actualRead + consume(actual)
+                    val otherSize = size + otherRead + consume(other)
+
+                    expected("to have the same content, but actual stream differs at pos $size. Actual stream: value=0x${actualBuffer[pos].toHexString()}, size=$actualSize. Other stream: value=0x${otherBuffer[pos].toHexString()}, size=$otherSize")
+                }
+            }
+
+        } else {
+
+            if(actualRead==0) {
+                val actualSize = actualRead + size
+                val otherSize = consume(other) + otherRead + size
+                expected("to have the same size, but actual stream size (${actualSize}) differs from other stream size ($otherSize)")
+            }
+            else if (otherRead==0) {
+                val actualSize = consume(actual) + actualRead + size
+                val otherSize = otherRead + size
+                expected("to have the same size, but actual stream size ($actualSize) differs from other stream size (${otherSize})")
+            }
+            else {
+                val pos = compare(Math.min(actualRead, otherRead), actualBuffer, otherBuffer)
+                val actualSize = consume(actual) + actualRead + size
+                val otherSize = consume(other) + otherRead + size
+                if(pos==null) {
+                    expected("to have the same size, but actual size (${actualSize}) differs from other size (${otherSize})")
+                }
+                else {
+                    expected("to have the same content, but actual stream differs at pos $size. Actual stream: value=0x${actualBuffer[pos].toHexString()}, size=$actualSize. Other stream: value=0x${otherBuffer[pos].toHexString()}, size=$otherSize")
+                }
+            }
+        }
+    }
+}
+
+/** Asserts that the actual stream has a different content as the other stream.
+ *
+ * @param other whichs content is compared to the actual one
+ */
+fun Assert<InputStream>.hasNotSameContentAs(other: InputStream) {
+    try {
+        hasSameContentAs(other)
```

Comment:
Similar to the comment above, `expected` might now throw an exception. You could extract the compare logic into it's own helper function that throws/returns and call them from the assertions.

Reply:
I assumed that expect leaves the method with an exception. I will add an appropriate return or break.

Reply:
I should document this somewhere, it's important to know for anyone writing assertions.